compiler/passes/cleanup.cpp

```diff
@@ -182,13 +182,19 @@ static void backPropagate(BaseAST* ast) {
 }
 
 static void handleNonTypedAndNonInitedVar(DefExpr* def) {
-  if (toVarSymbol(def->sym)) {
+  if (Symbol *sym = toVarSymbol(def->sym)) {
+    if(toShadowVarSymbol(sym)) {
```

Comment:
Probably these two ifs can be combined into `if (isShadowVarSymbol(def->sym))`

Reply:
I can change it to this:

`if (isVarSymbol(def->sym) && !isShadowVarSymbol(def->sym))`